docs/how-to-guides/running-feast-in-production.md

```diff
@@ -141,6 +199,147 @@ prediction = model.predict(feature_vector)
+
+## 5. Ingesting features from a Stream Source
+
+Recently Feast added functionality for [Stream Ingestion](../reference/alpha-stream-ingestion.md).
+Please note that this is still in an early phase and new incompatible changes may be introduced.
+
+### 5.1. Using Python SDK in your Apache Spark / Beam pipeline
+
+The default option to write features from a stream is to add Python SDK into your existing PySpark / Beam pipeline.
+Feast SDK provides writer implementation that can be called from `foreachBatch` stream writer in PySpark like this:
+
+```python
+store = FeatureStore(...)
+
+def feast_writer(spark_df):
+    pandas_df = spark_df.to_pandas()
+    store.write_to_online_store("driver_hourly_stats", pandas_df)
+
+streamingDF.writeStream.foreachBatch(feast_writer).start()
+```
+
+### 5.2. Push service *(still under development)*
+
+Alternatively, if you want to ingest features directly from a broker (eg, Kafka or Kinesis) you can use "Push Service", which will write to Online Store on push call.
+This service will expose an HTTP API or when deployed on Serverless platform as Lambda Function or Google Cloud Function,
+this service can be directly connected to Kinesis or PubSub.
+
+If you are using Kafka, [HTTP Sink](https://docs.confluent.io/kafka-connect-http/current/overview.html) could be utilized as middleware.
+In this case "Push Service" can be deployed on Kubernetes or as a Serverless function. 
+
+
+## 6. Monitoring
+
+Feast Services can report their metrics to the StatsD-compatible collector. To activate this function provide StatsD IP address and port when deploying helm chart (in future will be added to `feature_store.yaml`).
+
+We use [InfluxDB-style extension](https://github.com/prometheus/statsd_exporter#tagging-extensions) for StatsD format to be able to send tags along with metrics. Keep that in mind while selecting the collector ([telegraph](https://www.influxdata.com/blog/getting-started-with-sending-statsd-metrics-to-telegraf-influxdb/#introducing-influx-statsd) will work for sure).
+
+We chose StatsD since it's de-facto a standard with various implementations (eg, [1](https://github.com/prometheus/statsd_exporter), [2](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/statsd/README.md))
+and metrics could be easily exported to Prometheus, InfluxDB, AWS CloudWatch, etc.
+
+
+---
+
+Summarizing it all together we want to show several options of architecture that will be most frequently used in production:
+
+### Option #1 (currently preffered)
+
+* Feast SDK is being triggered by CI (eg, Github Actions). It applies the latest changes from Repository to Feast Registry
+* Airflow manages materialization jobs to ingest data from DWH to Online Store periodically
+* For stream ingestion Feast Python SDK is used in the existing Spark / Beam pipeline
+* Online Features are served with high performant Java Feature Server
```

Comment:
```suggestion
* Online Features are served via either a Python feature server or a high performance Java Feature Server
```

We do allow users to also integrate this into their existing services too though right? e.g. using the python SDK in a python server. I think our story is a little weaker on the java side if the user doesn't want to have a separate Java feature server (the Java SDK afaict wraps grpc calls to a feast serving instance)